### PR TITLE
vimPlugins.tsc-nvim: init at 2024-08-14

### PR DESCRIPTION
**Pull request by 347Online**
* Adds plugin [tsc.nvim](https://github.com/dmmulroy/tsc.nvim)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 347Online** [comment]
I ran the commands as outlined in the [docs](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/vim.section.md#adding-new-plugins-to-nixpkgs-adding-new-plugins-to-nixpkgs)

Is it kosher that it's making so many seemingly unrelated changes? I freshly rebased from upstream before running the update command

TL;DR is it normal to just update the vimPlugins in general when just adding a specific one?

**Review by PerchunPak — CHANGES_REQUESTED** [review]
> Is it kosher that it's making so many seemingly unrelated changes? I freshly rebased from upstream before running the update command
>
> TL;DR is it normal to just update the vimPlugins in general when just adding a specific one?

A feature for updating individual plugins is WIP, see https://github.com/NixOS/nixpkgs/pull/336137. For now, please remove all unrelated changes manually

This plugin depends on `tsc`, it is probably a good idea to patch it like here:
https://github.com/NixOS/nixpkgs/blob/3745f505bd20f65c2d71c345c919ef0fb83e1201/pkgs/applications/editors/vim/plugins/overrides.nix#L1489-L1495
Lines, where `tsc` is used:
https://github.com/dmmulroy/tsc.nvim/blob/82c37ebfe45d30763db6f45b54e18f1e485bb52c/lua/tsc/utils.lua#L19

**Comment by 347Online** [comment]
@PerchunPak Great feedback, thank you. I believe I've addressed the things you've raised, mind giving this another look?

**Review by PerchunPak — APPROVED** [review]
LGTM, just squash all commits into one please

**Comment by PerchunPak** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357769`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.tsc-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.tsc-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.tsc-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.tsc-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Comment by teto** [comment]
I squashed them from UI. Merging based on @PerchunPak 's approval
